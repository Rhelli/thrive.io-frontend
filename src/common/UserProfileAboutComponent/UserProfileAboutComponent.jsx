import React from 'react';
import PropTypes from 'prop-types';
import { userGenderDisplay, arrayDisplay } from '../../utils/userProfileUtils';
import styles from './UserProfileAboutComponent.module.scss';

const UserProfileAboutComponent = ({ userProfile }) => {
  const {
    about, gender, couple, pets, smoking, minBudget, maxBudget, areasLooking,
  } = userProfile;

  return (
    <div className={styles.userProfileAboutContainer}>
      <div>
        <div className={styles.userProfileAboutTopSection}>
          <h2>About</h2>
          <div className={styles.userProfileAbout}>
            <p>{about}</p>
          </div>
          <div className={styles.userProfileAreas}>
            <h2>Areas Looking</h2>
            <p>{arrayDisplay(areasLooking)}</p>
          </div>
        </div>
        <div className={styles.userProfileAboutBottomSection}>
          <div>
            <span>Budget:</span>
            <span>
              <p>
                £
                {minBudget}
              </p>
            </span>
            <span>
              <p> - </p>
            </span>
            <span>
              <p>
                £
                {maxBudget}
              </p>
            </span>
          </div>
          <div>
            <span>
              <p>Gender:</p>
            </span>
            <span className={styles.userProfileGenderBox}>
              <p>{userGenderDisplay(gender)[0]}</p>
              <p>{userGenderDisplay(gender)[1]}</p>
            </span>
          </div>
          <div>
            <span>
              <p>Couple?</p>
            </span>
            <span>
              <p>{couple}</p>
            </span>
          </div>
          <div>
            <span>
              <p>Smoking?</p>
            </span>
            <span>
              <p>{smoking}</p>
            </span>
          </div>
          <div>
            <span>
              <p>Pets?</p>
            </span>
            <span>
              <p>
                {arrayDisplay(pets)}
              </p>
            </span>
          </div>
        </div>
      </div>
    </div>
  );
};

UserProfileAboutComponent.propDefaults = {
  userProfile: {
    about: "This user hasn't added any information about themselves yet!",
    gender: 'No gender added yet.',
    couple: 'Unknown marital status.',
    pets: 'No pets information yet.',
    smoking: 'No smoking information yet',
    minBudget: 0,
    maxBudget: 0,
    areasLooking: 'No areas added yet.',
  },
};

UserProfileAboutComponent.propTypes = {
  userProfile: PropTypes.shape({
    about: PropTypes.string,
    gender: PropTypes.string,
    couple: PropTypes.string,
    pets: PropTypes.arrayOf(PropTypes.string),
    smoking: PropTypes.string,
    minBudget: PropTypes.number,
    maxBudget: PropTypes.number,
    areasLooking: PropTypes.arrayOf(PropTypes.string),
  }).isRequired,
};

export default UserProfileAboutComponent;
