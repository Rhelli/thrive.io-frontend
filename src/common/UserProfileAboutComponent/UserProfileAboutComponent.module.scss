.userProfileAboutContainer {
  width: 100%;
  padding: 20px;
}

.userProfileAboutTopSection {
  background-color: #ECE2FD;
  padding: 10px;
  margin-bottom: 20px;
  border-radius: 6px;

  h2 {
    padding-top: 10px;
  }

  .userProfileAbout {
    padding: 10px 0;
    line-height: 120%;
    border-bottom: 3px solid #D9C6FC;
  }

  .userProfileAreas {

    h2 {
      font-size: 18px;
      margin-bottom: 6px;
    }

    p {
      line-height: 120%;
    }
  }
}

.userProfileAboutBottomSection {
  display: flex;
  flex-flow: row wrap;
  justify-content: center;

  div {
    display: flex;
    flex-flow: row nowrap;
    background-color: #FCC6CD;
    border-left: 4px solid #F43651;
    border-radius: 4px;
    padding: 12px 12px;
    font-size: 14px;
    margin: 0 10px 10px 0;

    span:first-child {
      margin-right: 6px;
    }
  }

  .userProfileGenderBox {
    display: flex;
    flex-flow: row nowrap;

    p:first-child {
      margin-right: 4px;
    }
  }
}