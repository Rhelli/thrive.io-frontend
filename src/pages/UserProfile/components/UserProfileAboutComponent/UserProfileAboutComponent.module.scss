.userProfileAboutContainer {
  border: 1px solid orange;
  width: 100%;
  padding: 0 20px;
}

.userProfileAboutTopSection {
  h2 {
    margin: 10px 0;
  }

  .userProfileAbout {
    margin-bottom: 20px;
    line-height: 120%;
    border: 1px solid blue;
  }

  .userProfileAreas {
    border: 1px solid magenta;
    margin-bottom: 10px;

    p {
      line-height: 120%;
    }
  }
}

.userProfileAboutBottomSection {
  display: flex;
  flex-flow: row wrap;
  justify-content: center;

  div {
    display: flex;
    flex-flow: row nowrap;
    border: 1px solid red;
    border-radius: 4px;
    padding: 12px 12px;
    font-size: 14px;
    margin: 0 10px 10px 0;

    span:first-child {
      margin-right: 6px;
    }
  }

  .userProfileGenderBox {
    display: flex;
    flex-flow: row nowrap;

    p:first-child {
      margin-right: 4px;
    }
  }
}