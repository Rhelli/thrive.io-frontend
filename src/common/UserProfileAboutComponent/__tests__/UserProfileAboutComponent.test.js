import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import UserProfileAboutComponent from '../UserProfileAboutComponent';

Enzyme.configure({ adapter: new Adapter() });

const userProfile = {
  id: 3,
  name: 'Tonita',
  dob: '1936-08-19',
  email: 'thi@glover-rowe.org',
  userType: 'Looking',
  avatar: 'someimageurl.com',
  about: 'Sequi esse nesciunt ab.',
  occupation: 'Student',
  gender: 'Male',
  couple: 'Non-Couple',
  pets: ['Cats'],
  smoking: 'Smoking',
  minBudget: 1075,
  maxBudget: 1400,
  areasLooking: ['Dulwich', 'Camden'],
  advertiserType: null,
};

const partialUserProfile = {
  id: 3,
  name: 'Tonita',
  dob: '1936-08-19',
  email: 'thi@glover-rowe.org',
  userType: 'Looking',
  avatar: 'someimageurl.com',
  about: '',
  occupation: '',
  gender: '',
  couple: '',
  pets: [],
  smoking: '',
  minBudget: null,
  maxBudget: null,
  areasLooking: [],
  advertiserType: null,
};

const userProfileAboutComponent = (
  <UserProfileAboutComponent userProfile={userProfile} />
);

const partialProfileAboutComponent = (
  <UserProfileAboutComponent userProfile={partialUserProfile} />
);

describe('The appearance and functionality of the UserProfileAbout component', () => {
  const container = mount(userProfileAboutComponent);
  const partialContainer = mount(partialProfileAboutComponent);
  const bottomSection = container.find('.userProfileAboutBottomSection');
  const bottomSectionPartial = partialContainer.find('.userProfileAboutBottomSection');

  it('Matches the previous snapshot', () => {
    const tree = renderer.create(userProfileAboutComponent);
    expect(tree).toMatchSnapshot();
  });

  it('Has a master container with the correct attributes', () => {
    expect(container.childAt(0).hasClass('userProfileAboutContainer')).toBe(true);
  });

  it('Displays the profile top section with the provided prop info', () => {
    const aboutSection = container.find('.userProfileAboutTopSection');
    expect(aboutSection.childAt(0).text()).toContain('About');
    expect(aboutSection.childAt(1).text()).toBe('Sequi esse nesciunt ab.');
    expect(aboutSection.childAt(2).childAt(1).text()).toBe('Dulwich, Camden.');
  });

  it('Displays placeholder messages if there is no information provided', () => {
    const aboutSectionPartial = partialContainer.find('.userProfileAboutTopSection');
    expect(aboutSectionPartial.childAt(0).text()).toContain('About');
    expect(aboutSectionPartial.childAt(1).text()).toBe("This user hasn't added any information yet!");
    expect(aboutSectionPartial.childAt(2).childAt(1).text()).toBe('No areas added yet.');
  });

  it('Displays the profile bottom section with the provided info', () => {
    expect(bottomSection.childAt(0).childAt(1).text()).toBe('£1075');
    expect(bottomSection.childAt(0).childAt(3).text()).toBe('£1400');
    expect(bottomSection.childAt(1).childAt(0).text()).toBe('Gender:');
    expect(bottomSection.childAt(1).childAt(1).childAt(0).childAt(0).name()).toBe('FontAwesomeIcon');
    expect(bottomSection.childAt(1).childAt(1).childAt(1).text()).toBe('Male');
  });

  it('Displays placeholder messages if there is not information provided', () => {
    expect(bottomSectionPartial.childAt(0).childAt(1).text()).toBe('£tbd');
    expect(bottomSectionPartial.childAt(0).childAt(3).text()).toBe('£tbd');
    expect(bottomSectionPartial.childAt(1).childAt(0).text()).toBe('Gender:');
    expect(bottomSectionPartial.childAt(1).childAt(1).childAt(0).text()).toBe('tbd');
  });

  it('Displays the couple information correctly', () => {
    expect(bottomSection.childAt(2).childAt(0).text()).toBe('Couple?');
    expect(bottomSection.childAt(2).childAt(1).text()).toBe('Non-Couple');
  });

  it('Displays placeholder information if no couple info is available', () => {
    const bottomSectionPartial = partialContainer.find('.userProfileAboutBottomSection');
    expect(bottomSectionPartial.childAt(2).childAt(1).text()).toBe('tbd');
  });

  it('Displays the smoking information correctly', () => {
    expect(bottomSection.childAt(3).childAt(0).text()).toBe('Smoking?');
    expect(bottomSection.childAt(3).childAt(1).text()).toBe('Smoking');
  });

  it('Displays placeholder information if no smoking info is provided', () => {
    expect(bottomSectionPartial.childAt(3).childAt(1).text()).toBe('tbd');
  });

  it('Displays the pets information properly', () => {
    expect(bottomSection.childAt(4).childAt(0).text()).toBe('Pets?');
    expect(bottomSection.childAt(4).childAt(1).text()).toBe('Cats.');
  });

  it('Displays a placeholder if there is no pet information provided', () => {
    expect(bottomSectionPartial.childAt(4).childAt(1).text()).toBe('tbd');
  });
});
