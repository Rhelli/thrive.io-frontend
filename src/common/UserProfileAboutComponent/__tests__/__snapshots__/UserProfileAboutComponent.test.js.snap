// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance and functionality of the UserProfileAbout component Matches the previous snapshot 1`] = `
<div
  className="userProfileAboutContainer"
>
  <div>
    <div
      className="userProfileAboutTopSection"
    >
      <h2>
        About
      </h2>
      <div
        className="userProfileAbout"
      >
        <p>
          Sequi esse nesciunt ab.
        </p>
      </div>
      <div
        className="userProfileAreas"
      >
        <h2>
          Areas Looking
        </h2>
        <p>
          Dulwich, Camden.
        </p>
      </div>
    </div>
    <div
      className="userProfileAboutBottomSection"
    >
      <div>
        <span>
          Budget:
        </span>
        <span>
          <p>
            £
            1075
          </p>
        </span>
        <span>
          <p>
             
            -
             
          </p>
        </span>
        <span>
          <p>
            £
            1400
          </p>
        </span>
      </div>
      <div>
        <span>
          <p>
            Gender:
          </p>
        </span>
        <span
          className="userProfileGenderBox"
        >
          <p>
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-mars fa-w-12 "
              data-icon="mars"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M372 64h-79c-10.7 0-16 12.9-8.5 20.5l16.9 16.9-80.7 80.7c-22.2-14-48.5-22.1-76.7-22.1C64.5 160 0 224.5 0 304s64.5 144 144 144 144-64.5 144-144c0-28.2-8.1-54.5-22.1-76.7l80.7-80.7 16.9 16.9c7.6 7.6 20.5 2.2 20.5-8.5V76c0-6.6-5.4-12-12-12zM144 384c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </p>
          <p>
            Male
          </p>
        </span>
      </div>
      <div>
        <span>
          <p>
            Couple?
          </p>
        </span>
        <span>
          <p>
            Non-Couple
          </p>
        </span>
      </div>
      <div>
        <span>
          <p>
            Smoking?
          </p>
        </span>
        <span>
          <p>
            Smoking
          </p>
        </span>
      </div>
      <div>
        <span>
          <p>
            Pets?
          </p>
        </span>
        <span>
          <p>
            Cats.
          </p>
        </span>
      </div>
    </div>
  </div>
</div>
`;
